core/auth/src/main/java/software/amazon/awssdk/auth/token/credentials/ProfileTokenProvider.java

```diff
@@ -137,6 +123,29 @@ public void close() {
         IoUtils.closeIfCloseable(tokenProvider, null);
     }
 
+    private SdkTokenProvider createTokenProvider(ProfileFile profileFile, String profileName) {
+        // Load the profile and token provider
+        return profileFile.profile(profileName)
+                          .flatMap(p -> new ProfileTokenProviderLoader(profileFile, p).tokenProvider())
+                          .orElseThrow(() -> {
+                              String errorMessage = String.format("Profile file contained no information for " +
+                                                                  "profile '%s': %s", profileName, profileFile);
+                              return SdkClientException.builder().message(errorMessage).build();
+                          });
+    }
+
+    private void handleProfileFileReload(ProfileFile profileFile) {
+        tokenProvider = createTokenProvider(profileFile, profileName);
```

Comment:
Does this need to be synchronized?

Reply:
@zoewangg? If needed, will have to make same change to `ProfileCredentialsProvider`.